test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -3,61 +3,90 @@
 require 'dns_dnscmd/dns_dnscmd_main'
 
 class DnsCmdTest < Test::Unit::TestCase
+  def setup
+    Proxy::Dns::Plugin.load_test_settings(
+        :dns_server => 'a_server',
+        :dns_zones => ['bar.domain.local', 'domain.local', '33.168.192.in-addr.arpa', '168.192.in-addr.arpa'])
+
+    to_parse = '
+Enumerated zone list:
+        Zone count = 8
+
+ Zone name                      Type       Storage         Properties
+
+ .                              Cache      AD-Domain
+ _msdcs.bar.domain.local        Primary    AD-Forest       Secure
+ 168.192.in-addr.arpa           Primary    AD-Domain       Rev
+ 33.168.192.in-addr.arpa        Primary    AD-Domain       Secure Rev Aging
+ e.e.d.8.b.d.0.1.0.0.2.ip6.arpa Primary    AD-Domain       Secure Rev
+ bar.domain.local               Primary    AD-Domain       Secure Aging
+ domain.local                   Primary    AD-Domain       Secure
+ domain.com                     Secondary    File
+ TrustAnchors                   Primary    AD-Forest
+
+
+Command completed successfully.'.split("\n")
+    Proxy::Dns::Dnscmd::Record.stubs(:execute).with('/EnumZones').returns(to_parse)
```

Comment:
You actually don't need to stub Proxy::Dns::Dnscmd::Record#execute. Instead:

```
class DnscmdForTesting < Proxy::Dns::Dnscmd::Record
  def initialize(dns_zones)
    @enum_dns_zones = dns_zones
  end
  attr_accessor : enum_dns_zones
end
```

then for tests that are validating logic in methods *other* than Proxy::Dns::Dnscmd::Record.#enum_dns_zones:

```
server = DnscmdForTesting.new(['domain.com', 'domain.local' ...])
```  

Alternatively, you can move the line above into ```setup``` method.